src/fixup.js

```diff
@@ -139,4 +140,78 @@
+  var request = new XMLHttpRequest();
+  request.open('GET', '//www.w3.org/TR/tr-outdated-spec.json', true);
+  request.onload = function() {
+    if (request.status >= 200 && request.status < 400) {
+      var outdatedSpecs = JSON.parse(request.responseText);
+      const pathname = window.location.pathname;
+      var spec = Object.keys(outdatedSpecs).filter(function(k) {return pathname.indexOf(k) === 0;}).shift();
+      if (spec) {
+        var css = 'a#deprecationnote:hover{ background-color: transparent }'
+        ,   style = document.createElement('style');
+        if (style.styleSheet) {
+          style.styleSheet.cssText = css;
+        } else {
+          style.appendChild(document.createTextNode(css));
+        }
+        document.getElementsByTagName('head')[0].appendChild(style);
+        var nodes = document.querySelectorAll("body, h1, h2, h3");
+        for (var i = 0; i < nodes.length; i++) {
+          nodes[i].setAttribute("style", "background-color: rgba(0,0,0,0.5);");
+        }
+        var node = document.createElement("p");
+        node.className = "outdatedwarning";
+        var outdatedValue = "bottom:50%; left:0; right:0; margin:0 auto 0 auto; width:50%; background:maroon; color:white; border-radius:1em; box-shadow:0 0 1em red; padding:2em; text-align:center; z-index:2;";
+        node.style.cssText = "position:fixed; " + outdatedValue;
+
+        var warning = '<strong>This version is outdated!</strong><div>For the latest version, please look at the <a id="deprecationnote" style="color:white" href="' + outdatedSpecs[spec] + '"> ' + outdatedSpecs[spec] + '</a>.</div><input onclick="collapseWarning(false)" style="margin:0; border:0; padding:0.25em 0.5em; background:transparent; color:black; position:absolute; top:0em; right:0; font:1.25em sans-serif; text-align:center;" type="button" value="&#9662; collapse">';
+        node.innerHTML = warning;
+
+        document.querySelector("body").appendChild(node);
+
+        /* Apply a slightly different style for print media */
+        var mediaQueryList = window.matchMedia('print');
+        mediaQueryList.addListener(function(mql) {
+          if (mql.matches) {
+            node.style.cssText = "position:absolute; border-style:solid; border-color:red; " + outdatedValue;
+            node.innerHTML = '<strong>This version is outdated!</strong><div>For the latest version, please look at the <a id="deprecationnote" style="color:white" href="' + outdatedSpecs[spec] + '"> ' + outdatedSpecs[spec] + '</a>.</div>';
+          } else {
+            node.style.cssText = "position:fixed; " + outdatedValue;
+            node.innerHTML = warning;
+          }
+        });
+      }
+    } else {
+      console.error("Error downloading https://www.w3.org/TR/tr-outdated-spec.json");
+    }
+  };
+
+  request.onerror = function() {
+    console.error("Error downloading https://www.w3.org/TR/tr-outdated-spec.json");
+  };
+
+  request.send();
 })();
+
+function collapseWarning(details) {
```

Comment:
You should get this for free from `<details>` and then just use CSS to inline it. See how we do it in:

https://s3.amazonaws.com/pr-preview/markalanrichards/payment-request/patch-1.html

Which I don't think requires any JS.